### ISSUE TITLE
Update staticcheck for false positive in false positive in SA4000

### ISSUE THREAD

**Issue by kortschak** [golangci/golangci-lint#829]
https://github.com/dominikh/go-tools/issues/623 has been fixed. So this change should probably be pulled into golangci-lint.

```
~/src/github.com/golangci/golangci-lint $ golangci-lint --version
golangci-lint has version (devel) built from (unknown, mod sum: "") on (unknown)
~/src/github.com/golangci/golangci-lint $ git tag -l --points-at HEAD
v1.21.0
```

```
~/src/gonum.org/v1/gonum/graph/path/dynamic $ golangci-lint run
dstarlite.go:412:39: SA4000: identical expressions on the left and right side of the '!=' operator (staticcheck)
func (k key) isBadKey() bool { return k != k }
```

**Comment by kortschak** [comment]
This appears to have been done.
